### PR TITLE
refactor: fetch reward min blocks

### PR DESCRIPTION
**Pull request by andreabadesso**
Fixes https://github.com/HathorNetwork/hathor-wallet-service-sync_daemon/issues/95

### Acceptance Criteria
- We should fetch the reward_min_blocks from the fullnode and use it instead of an env variable

### Security Checklist
- [X] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

File: packages/daemon/src/config.ts [hunk 1]
```diff
@@ -48,13 +50,18 @@ export const DB_USER = process.env.DB_USER;
...
 
+// Other
+export const USE_SSL = process.env.USE_SSL;
+
 
 export default () => ({
   SERVICE_NAME,
   CONSOLE_LEVEL,
   TX_CACHE_SIZE,
   BLOCK_REWARD_LOCK,
```
Comment (glevco) [inline packages/daemon/src/config.ts]: Should we remove this, and its definition above?

File: packages/daemon/src/utils/helpers.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { USE_SSL, FULLNODE_HOST } from '../config';
+
+export const getFullnodeHttpUrl = () => {
+  const protocol = USE_SSL ? 'https://' : 'http://';
+
+  return `${protocol}${FULLNODE_HOST}/v1a/`;
```
Comment (r4mmer) [inline packages/daemon/src/utils/helpers.ts]: If the wallet-service runs on nodejs we can use `path.join` to safely join the url parts.

**Review by r4mmer — COMMENTED** [review]
The data from the fullnode is used with the name `rewardMinBlocks` but it is set on the initial state as `minRewardBlocks`, but I could not find where one is converted to the other.

File: packages/daemon/src/config.ts (hunk 1 above)
Comment (andreabadesso) [reply packages/daemon/src/config.ts]: Yes we should, done in f4c2fd24fae2068e1961b52012015fb90f27b8a2

File: packages/daemon/src/utils/helpers.ts (hunk 2 above)
Comment (andreabadesso) [reply packages/daemon/src/utils/helpers.ts]: Updated it to use URL in dfe48c88978f587fad943ad167571f7a438e694f

**Comment by andreabadesso** [comment]
> The data from the fullnode is used with the name `rewardMinBlocks` but it is set on the initial state as `minRewardBlocks`, but I could not find where one is converted to the other.

Refactored it to use `rewardMinBlocks` in 7c4e900b71a815bc8b17e15653b3675d7cc7030e

**Review by r4mmer — APPROVED** [review]

File: packages/daemon/src/config.ts (hunk 1 above)
Comment (glevco) [reply packages/daemon/src/config.ts]: Thanks!

**Review by glevco — APPROVED** [review]

**Pull request merged**